runtime/storage/cache/cache.go

```diff
@@ -0,0 +1,267 @@
+// Package cache provides the ability to define distributed Redis cache clusters
+// and functionality to use them in a fully type-safe manner.
+//
+// For more information see https://encore.dev/docs/develop/caching
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// ClusterConfig represents the configuration of cache clusters.
+type ClusterConfig struct {
+	// EvictionPolicy decides how the cache evicts existing keys
+	// to make room for new data when the max memory limit is reached.
+	//
+	// If not specified the cache defaults to AllKeysLRU.
+	EvictionPolicy EvictionPolicy
+}
+
+// An EvictionPolicy describes how the cache evicts keys to make room for new data
+// when the maximum memory limit is reached.
+//
+// See https://redis.io/docs/manual/eviction/#eviction-policies for more information.
+type EvictionPolicy string
+
+// NOTE: These values need to be added to the runtimeconstants package
+// and the parser package for the parser to be aware of them.
+
+const (
```

Comment:
I also don't like how it feels like we've squashed `AllKeys` / `Volatile` with `LRU` / `LFU` / `Random` / `NoEviction`.

These feel like two different settings - where the documentation for each setting can be read in isolation. The only special edge case is simply if `NoEvicition` is set it doesn't matter about `Volitle` vs `AllKeys`.

In this case then `Volatile` could be called `KeysWithExpiry`. So the option becomes

```go
cache.New(..., cache.Options{
  EvictionPolicy: cache.EvicationPolicy {
    Type: cache.Randomly, // cache.LeastFrequentlyUsed / cache.LeastRecentlyUsed / cache.NoEviction
    What:  cache.AllKeys // cache.KeysWithExpiries
  }
})
```